lnwallet/channel.go

```diff
@@ -1825,6 +1943,75 @@ func (lc *LightningChannel) restoreStateLogs(
 		}
 	}
 
+	// Restore dangling remote log updates that we haven't signed for yet.
+	err := lc.restorePendingRemoteUpdates(
+		danglingRemoteUpdates, localCommitment.height,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// restorePendingRemoteUpdates restores the remote log updates that we haven't
+// yet signed for.
+func (lc *LightningChannel) restorePendingRemoteUpdates(
+	danglingRemoteUpdates []channeldb.LogUpdate,
+	localCommitmentHeight uint64) error {
+
+	if len(danglingRemoteUpdates) == 0 {
+		lc.log.Debugf("No dangling remote updates to restore")
+		return nil
+	}
+
+	lc.log.Debugf("Restoring %v dangling remote updates",
+		len(danglingRemoteUpdates))
+
+	var (
+		lastHtlcIndex uint64
+		containsHtlc  bool
+	)
+	for _, logUpdate := range danglingRemoteUpdates {
+		logUpdate := logUpdate
+
+		payDesc := lc.remoteLogUpdateToPayDesc(
+			&logUpdate, lc.localUpdateLog, localCommitmentHeight,
+		)
+
+		switch payDesc.EntryType {
+		case Add:
+			lc.remoteUpdateLog.restoreHtlc(payDesc)
```

Comment:
some adds will be restored twice now, both when checking incoming HTLCs on our local commit, and here. But maybe that's okay?

Reply:
hmm... maybe it isn't necessary then to save and restore add msgs?

Reply:
Thought about this a bit more. The operation is idempotent, so it doesn't seem to hurt. But maybe we do need the save/restore for add msgs in case an htlc is added and later settled on the local commit. If we aren't able to sign during that time, we still need to process those messages later.